### PR TITLE
Fix lints for system proofs [berk]

### PR DESCRIPTION
**Pull request by dkijania**
Fixed lints for system proofs commit ancestor. Currently script takes into consideration lack of base branch setting and default condition

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Review by shimkiv — APPROVED** [review]
LGTM

**Review by rbonichon — APPROVED** [review]

**Comment by dkijania** [comment]
!ci-build-me

File: scripts/check-proof-systems-submodule.sh [hunk 1]
```diff
@@ -10,11 +10,20 @@ git config http.sslVerify false
...
+  exit 1
+fi
+
 declare -A BRANCH_MAPPING=(
   ["rampup"]="compatible" 
   ["berkeley"]="berkeley"
   ["develop"]="develop"
   ["izmir"]="master"
+  # Default case
+  ["${BUILDKITE_PULL_REQUEST_BASE_BRANCH}"]="${BUILDKITE_PULL_REQUEST_BASE_BRANCH}"
```
Comment (mrmr1993) [inline scripts/check-proof-systems-submodule.sh:26]: I don't think this is the desired default. We should probably default to `master`

**Review by mrmr1993 — CHANGES_REQUESTED** [review]

File: scripts/check-proof-systems-submodule.sh (hunk 1 above)
Comment (rbonichon) [reply scripts/check-proof-systems-submodule.sh:26]: Defaulting would make this part fail when running the CI in a branch B targeting a branch different from master/develop/berkeley. This happened for example in the Pickles fixes PR train, where you took care of having matching branch names in proof-systems/snarkyjs.. So I kind of thought this wouldn't be bad discipline to enforce that.

**Comment by dkijania** [comment]
!ci-build-me

File: buildkite/src/Jobs/Lint/Fast.dhall [hunk 2]
```diff
@@ -32,7 +31,10 @@ in  Pipeline.build
         , dirtyWhen = [
             S.strictly (S.contains "Makefile"),
             S.strictlyStart (S.contains "src/"),
```
Comment (mrmr1993) [inline buildkite/src/Jobs/Lint/Fast.dhall:33]: You need to remove the trailing `,` here

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
Change implementation according do @mrmr1993 idea:

> What I'm imagining is a script that takes the proof-systems branch name, and then we run 4 CI jobs on every PR
./merged-to-proof-systems compatible
./merged-to-proof-systems berkeley
./merged-to-proof-systems develop
./merged-to-proof-systems main
that runs the same check as currently, but with the given argument as the target branch. Then, we can change branch protection rules to match (e.g. PRs against berkeley must be pass the berkeley job

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Pull request merged**